lib/job/arBaseJob.class.php

```diff
@@ -401,4 +487,87 @@ protected function canBeFullyExecuted()
     // If this job is the first one, it can be fully executed
     return $this->job->id === $runningJobs[0]['id'];
   }
+
+  /**
+   * Return an array of digital object ids if any are attached to clipboard items
+   * and current user has permission to view masters
+   *
+   * @return array
+   */
+  protected function getDigitalObjects()
+  {
+    // Prepare array for digital object ids
+    $digitalObjects = array();
+
+    // Process if export option is set and this is a description or actor export
+    if (sfConfig::get('app_clipboard_export_digitalobjects_enabled', false) && ('informationObject' == $this->params['objectType'] || 'actor' == $this->params['objectType']))
+    {
+      // Get clipboard objects
+      $criteria = new Criteria;
+
+      // Filter on clipboard slugs
+      $criteria->add(QubitSlug::SLUG, $this->params['params']['slugs'], Criteria::IN);
+
+      switch ($this->params['objectType'])
+      {
+        case 'informationObject':
+          $criteria->addJoin(QubitInformationObject::ID, QubitSlug::OBJECT_ID);
+          // Hide drafts if necessary
+          if(!$this->params['public'])
```

Comment:
Should this be the other way around? If it's public, then filter drafts?

Reply:
Nice catch. Fixed.